### PR TITLE
Reset splom selectBatch and unselectBatch on updates

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/4588

before: https://codepen.io/etpinard/pen/zYGKbVv (try selecting points, then selecting no points)
after: https://codepen.io/etpinard/pen/yLNareB (this one should reset the selected points)

cc @archmoj

### CODE REVIEW

File: test/jasmine/tests/splom_test.js [hunk 1]
```diff
@@ -1837,4 +1837,45 @@ describe('Test splom select:', function() {
...
+        }))
+        .then(function() { return _select([[50, 50], [195, 195]]); })
+        .then(_assert('after selection', {
+            selectBatch: [1],
+            unselectBatch: [0, 2]
+        }))
+        .then(function() { return Plotly.restyle(gd, 'selectedpoints', null); })
+        .then(_assert('after API clear', {
+            selectBatch: [],
+            unselectBatch: []
```
Comment (etpinard) [inline test/jasmine/tests/splom_test.js]: Before, these used to remain `[1]` and `[0, 2]` respectively.

**Comment by archmoj** [comment]
Well done. :dancer:
This PR closes 2000th plotly.js issue!
:1st_place_medal:

**Comment by etpinard** [comment]
image

haha, good eye!

**Comment by archmoj** [comment]
@etpinard 
Next stop 2020! :checkered_flag:

**Pull request merged**